File: frontend/src/types/budget.ts
/**
 * @file types/budget.ts
 * @description 预算管理类型定义
 */
export interface Budget {
  id?: number;
  budgetYear: number;
  deptId?: number;
  categoryId?: number;
  budgetType: 'PURCHASE' | 'MAINTENANCE' | 'OPERATION';
  totalAmount: number;
  usedAmount: number;
  committedAmount: number;
  status: 'DRAFT' | 'APPROVED' | 'CLOSED';
  approvedBy?: number;
  createTime?: string;
  updateTime?: string;
  deptName?: string;
  categoryName?: string;
}

export interface BudgetQuery {
  budgetYear?: number;
  deptId?: number;
  categoryId?: number;
  budgetType?: string;
  status?: string;
  page?: number;
  pageSize?: number;
}

export interface ExecutionRate {
  budgetId: number;
  budgetYear: number;
  budgetType: string;
  totalAmount: number;
  usedAmount: number;
  committedAmount: number;
  executionRate: number;
  status: string;
}

export interface OverBudgetAlert {
  budgetId: number;
  budgetYear: number;
  budgetType: string;
  totalAmount: number;
  usedAmount: number;
  overshoot: number;
  status: string;
}

File: backend/src/main/java/com/ams/entity/Budget.java
package com.ams.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("budget")
public class Budget implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Integer budgetYear;
    private String tenantId;
    private Long deptId;
    private Long categoryId;
    private String budgetType;
    private BigDecimal totalAmount;
    private BigDecimal usedAmount;
    private BigDecimal committedAmount;
    private String status;
    private Long approvedBy;
    private Long createBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}
